src/abstract.jl

```diff
@@ -216,61 +235,81 @@ function addcuts!{N, T}(man::AbstractCutPruner{N, T},
             man.b[R] = br
-            replacecuts!(man, R, mycutr)
+            replacecuts!(man, R, Ar, br, mycutr)
         end
     end
-    if !isempty(b)
-        # Just append cuts
-        status = ncur + (1:nnew)
-        man.A = [man.A; A]
-        man.b = [man.b; b]
-        pushcuts!(man, mycut)
-    end
 
     status
 end
 
-"""Keep only cuts whose indexes are in Vector `K`."""
-function keeponly!(man::AbstractCutPruner, K::AbstractVector{Int})
+# Low-level functions
+function _keeponlycuts!(man::AbstractCutPruner, K::AbstractVector{Int})
     man.A = man.A[K, :]
     man.b = man.b[K]
-    man.trust = man.trust[K]
+    man.ids = man.ids[K]
+    if hastrust(man)
+        man.trust = gettrust(man)[K]
+    end
 end
 
-"""Get a Vector of Float64 specifying the initial trusts of `mycut`."""
-function initialtrusts(man::AbstractCutPruner, mycut::AbstractVector{Bool})
-    Float64[initialtrust(man, mc) for mc in mycut]
+"""Keep only cuts whose indexes are in Vector `K` in CutPruner `man`. If `K` is not sorted, the cuts will change their order accordingly."""
+function keeponlycuts!(man::AbstractCutPruner, K::AbstractVector{Int})
+    _keeponlycuts!(man, K)
+end
+
+"""Remove cuts whose indexes are in Vector `K` in CutPruner `man`."""
+function removecuts!(man::AbstractCutPruner, K::AbstractVector{Int})
+    keeponlycuts!(man, setdiff(1:ncuts(man), K))
+end
+
+function _replacecuts!(man::AbstractCutPruner, K::AbstractVector{Int}, A, b)
+    man.A[K, :] = A
+    man.b[K] = b
+    man.ids[K] = newids(man, length(K))
 end
 
-"""Reset trust of cuts with indexes in `js`."""
-function replacecuts!(man::AbstractCutPruner, js::AbstractVector{Int}, mycut::AbstractVector{Bool})
-    man.trust[js] = initialtrusts(man, mycut)
-    man.ids[js] = newids(man, length(js))
+"""Replace cuts at indexes in `K` by cuts in (A, b, mycut) in CutPruner `man`."""
+function replacecuts!(man::AbstractCutPruner, K::AbstractVector{Int}, A, b, mycut::AbstractVector{Bool})
+    _replacecuts!(man, K, A, b)
+    man.trust[K] = initialtrusts(man, mycut)
 end
 
-function pushcuts!(man::AbstractCutPruner, mycut::AbstractVector{Bool})
+function _appendcuts!(man::AbstractCutPruner, A, b)
+    man.A = [man.A; A]
+    man.b = [man.b; b]
+    append!(man.ids, newids(man, length(b)))
+end
+
+"""Append cuts (A, b, mycut) in CutPruner `man`."""
+function appendcuts!(man::AbstractCutPruner, A, b, mycut::AbstractVector{Bool})
+    _appendcuts!(man, A, b)
     append!(man.trust, initialtrusts(man, mycut))
-    append!(man.ids, newids(man, length(mycut)))
 end
 
+# Unexported utilities
+
 function newids(man::AbstractCutPruner, n::Int)
     (man.id+1):(man.id += n)
 end
 
+"""Get a Vector of Float64 specifying the initial trusts of `mycut`."""
+function initialtrusts(man::AbstractCutPruner, mycut::AbstractVector{Bool})
+    Float64[initialtrust(man, mc) for mc in mycut]
+end
+
 function normalizedcut{T}(A::AbstractMatrix{T}, b::AbstractVector{T}, k::Int, isfun::Bool, tol::Float64)
```

Comment:
It is not exported and only used from the inside in which case we always have a value for `tol` so I wouldn't see which leaving a default value would help. Moreover if we forget to give `tol` then it won't fail and it will be a hidden bug :(